pyresample/test/test_bucket.py

```diff
@@ -174,10 +174,61 @@ def test_get_sum_nan_data_skipna_true(self):
         result = self._get_sum_result(data, skipna=True)
         # 2 + nan is 2
         self.assertEqual(np.count_nonzero(result == 2.), 1)
+        # 5 is untouched in a single bin
+        self.assertEqual(np.count_nonzero(result == 5.), 1)
         # all-nan and rest is 0
         self.assertEqual(np.count_nonzero(np.isnan(result)), 0)
         self.assertEqual(np.nanmin(result), 0)
 
+    def test_get_sum_non_default_fill_value_skipna_false(self):
+        """Test drop-in-a-bucket sum for data input with non-default fill_value and skipna=False."""
+        data = da.from_array(np.array([[2., 255], [5., 255]]),
+                             chunks=self.chunks)
+
+        result = self._get_sum_result(data, skipna=False, fill_value=255)
+        # 2 + fill_value  is fill_value, all-fill_value is fill_value
+        self.assertEqual(np.count_nonzero(result == 255), 2)
+        # 5 is untouched in a single bin
+        self.assertEqual(np.count_nonzero(result == 5.), 1)
+        # rest is 0
+        self.assertEqual(np.nanmin(result), 0)
```

Comment:
These tests all seem very similar. Could you separate these tests into a pytest-based class and use parametrization to get all the cases covered in a single test? Bonus points if you update this existing class to pytest (resampler fixture, remove setup method, remove base class, use `assert` instead of `self.assertX`).

Reply:
Done, and I like bonus points so I moved everything to pytest :)